backend/coins/eth/coin.go

```diff
@@ -42,64 +39,46 @@ type TransactionsSource interface {
 		[]*accounts.TransactionData, error)
 }
 
-// TransactionsSourceMaker creates a transaction source.
-type TransactionsSourceMaker func() TransactionsSource
-
-// TransactionsSourceEtherScan creates a etherscan transactions source maker.
-func TransactionsSourceEtherScan(etherScanURL string, httpClient *http.Client) TransactionsSourceMaker {
-	return func() TransactionsSource { return etherscan.NewEtherScan(etherScanURL, httpClient) }
-}
-
-// TransactionsSourceNone is used if no transactions source should be used.
-var TransactionsSourceNone TransactionsSourceMaker = func() TransactionsSource { return nil }
-
 // Coin models an Ethereum coin.
 type Coin struct {
 	observable.Implementation
-	initOnce              sync.Once
 	client                rpcclient.Interface
 	code                  coin.Code
 	unit                  string
 	feeUnit               string
 	net                   *params.ChainConfig
 	blockExplorerTxPrefix string
-	nodeURL               string
 	erc20Token            *erc20.Token
 
-	makeTransactionsSource TransactionsSourceMaker
-	transactionsSource     TransactionsSource
-	httpClient             *http.Client
+	transactionsSource TransactionsSource
 
 	log *logrus.Entry
 }
 
 // NewCoin creates a new coin with the given parameters.
-// makeTransactionsSource: provide `TransactionsSourceNone` or `TransactionsSourceEtherScan()`.
+// transactionsSource: can be nil, in which case transactions will not be shown.
```

Comment:
... shown somewhere in the UI?
maybe a better place for this comment line is where the coin and its transactionSource is used.

Reply:
I elaborated the comment a bit.